### ISSUE TITLE
[fix] password field label must be correctly programmatically associated on auth sign in template example

### ISSUE THREAD

**Issue by shkeating** [trussworks/react-uswds#2610]
**ReactUSWDS Version: 5.4.0 & USWDS Version: 3.6.0**

# Describe the bug

In our storybook example of the [auth sign in page template ](https://trussworks.github.io/react-uswds/?path=/story/page-templates-authentication-pages--sign-in) we do not correctly associate the label of the password field. There is a `label` present, however, it has a `for=` attribute with the id of another example in the file.

<img width="1320" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/ec992894-a038-4a78-ad8a-c87a2a24e44f">

<img width="1327" alt="image" src="https://github.com/trussworks/react-uswds/assets/59394696/49a5d5c3-2b9f-439c-96c7-df0ab59f33c0">

Currently the `for` attribute is associated to an `id` of `email`, when our password field has an `id` of `password-sign-in`. This could lead to a confusing page experience for users of assistive technology when using a site that may use this page template's markup as a reference.

# To Reproduce

1. Go to [page template for authentication sign in](https://trussworks.github.io/react-uswds/iframe.html?args=&id=page-templates-authentication-pages--sign-in&viewMode=story)
2. run an automated accessibility check
3. see test failure for no label on the password field

# Expected behavior

The label's `for` attribute must match the id of the input it is associated with. In this case, we want our label to have `for="email"` to be `for="password-sign-in"` instead. this was likely overlooked int he initial implementation of this page.

# Additional context

- [Labelling Controls - W3C](https://www.w3.org/WAI/tutorials/forms/labels/)
- file(s) to be edited: `src/stories/templates/signin.stories.tsx`

**Comment by aybeedee** [comment]
Hi @shkeating, I'd like to work on this, can I kindly be assigned?

**Comment by shkeating** [comment]
@aybeedee absolutely, go for it!!
